components/login/provider/index.jsx

```diff
@@ -31,17 +33,30 @@ import { generateRedirectUrl } from "../../../src/windowHelpers";
 
 import getIdentityProviders from "../../../constants/provider";
 import styles from "./styles";
+import useIdpFromQuery from "../../../src/hooks/useIdpFromQuery";
 
 const providers = getIdentityProviders();
 const TESTCAFE_ID_LOGIN_TITLE = "login-title";
 const TESTCAFE_ID_LOGIN_BUTTON = "login-button";
+export const TESTCAFE_ID_LOGIN_FIELD = "login-field";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
 export default function Provider() {
   const bem = useBem(useStyles());
-  const [providerIri, setProviderIri] = useState("https://inrupt.net");
   const { login } = useSession();
+  const idp = useIdpFromQuery();
+  const [providerIri, setProviderIri] = useState("https://inrupt.net");
+  const loginFieldRef = createRef();
+
+  useEffect(() => {
+    if (idp) {
+      setProviderIri(idp.iri);
+    }
+    if (idp && loginFieldRef.current) {
```

Comment:
`loginFieldRef.current` could be moved below:

`loginFieldRef.current?.querySelector("input")?.focus();`